master/internal/trial.go

```diff
@@ -619,8 +620,15 @@ func (t *trial) handleAllocationExit(exit *task.AllocationExited) error {
 			})
 		}
 
-		// TODO(DET-9897) redo capacity check when we decide to allocate again.
-		// Since we could have excluded nodes.
+		blockedNodes, err := logpattern.GetBlockedNodes(context.TODO(), t.taskID)
+		if err != nil {
+			return err
+		}
+		if len(blockedNodes) > 0 {
+			if err := t.checkResourcePoolRemainingCapacity(); err != nil {
+				return err
```

Comment:
how did we test this? i don't think just returning an error is quite right here. i think like all the code around it, we need to transition to a terminal state.

Reply:
there is an invariant here we need to have commented so folks don't miss it (cc @erikwilson ). `handleAllocationExit` must transition to a terminal state or reallocate, otherwise the system is in an invalid state.